zebra-consensus/src/block/check.rs

```
@@ -25,3 +30,40 @@ pub fn is_coinbase_first(block: &Block) -> Result<(), Error> {
     }
     Ok(())
 }
+
+/// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
+pub fn is_subsidy_correct(block: &Block) -> Result<(), Error> {
+    let height = block.coinbase_height().unwrap();
+
+    let coinbase = block.transactions.get(0).ok_or("no coinbase transaction")?;
+    let outputs = coinbase.outputs();
+
+    // Todo: we need the network here.
+    let network = Network::Mainnet;
+
+    let canopy_height = Canopy.activation_height(network).ok_or("no canopy")?;
+    if height >= canopy_height {
+        // dont validate canopy yet
+        return Ok(());
+    }
```

I feel like it might be better to panic here rather than letting it silently pass

---

Done at https://github.com/ZcashFoundation/zebra/pull/1051/commits/b5b2877f8aee99b17e30d23e21149d26974a223c